### PR TITLE
feat(rest): new endpoint `/moderationrequest`

### PR DESCRIPTION
**Pull request by GMishx**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

The PR adds new endpoint `/moderationrequest` to allow listing ModerationRequests from REST API.

* Which issue is this pull request belonging to and how is it solving it?
> PR is related to issue #1849
* Did you add or update any new dependencies that are required for your change?
> No

Issue: #1849

Reviewer can try to load the following implemented endpoints:
- `GET` `/moderationrequest`
  - Should respond with all moderation requests available to the user.
  - The endpoint should be paginated.
- `GET` `/moderationrequest/{id}`
  - Should respond with single moderation request.
  - Additionally, should embed the requesting user's object and relevant project/component/release/license.
- `GET` `/moderationrequest/byState?state={state}`
  - Should filter only moderation requests with given state.
  - State should be checked against possible values of `open` & `closed`. Other values should throw 400 bad request.
> Have you implemented any additional tests?

No

### Checklist
Must:
- [x] All related issues are referenced in commit messages and in PR

### CODE REVIEW

**Comment by rudra-superrr** [comment]
Hi @GMishx ,
Testing was successful but Pagination details are not getting displayed in response structure for endpoint /byState.

**Comment by GMishx** [comment]
> Hi @GMishx , Testing was successful but Pagination details are not getting displayed in response structure for endpoint /byState.

Hello @rudra-superrr , I have updated the `/byState` endpoint to send the pagination object in response. Please test again.

**Comment by rudra-superrr** [comment]
Testing was successful.

File: backend/src/src-moderation/src/main/java/org/eclipse/sw360/moderation/ModerationHandler.java [hunk 1]
```diff
@@ -316,12 +323,20 @@ public Map<String, Long> getCountByModerationState(User user) throws TException
 
     @Override
     public Map<PaginationData, List<ModerationRequest>> getRequestsByModeratorWithPagination(User user,
-            PaginationData pageData, boolean open) throws TException {
+                                                                                             PaginationData pageData, boolean open) throws TException {
```
Comment (rudra-superrr) [inline backend/src/src-moderation/src/main/java/org/eclipse/sw360/moderation/ModerationHandler.java:326]: Please remove extra spaces on lines 326 and 334.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/JacksonCustomizations.java [hunk 2]
```diff
@@ -236,7 +250,7 @@ static abstract class ProjectMixin extends Project {
             abstract public String getClearingRequestId();
         }
 
-	static abstract class EmbeddedProjectMixin extends ProjectMixin {
+	    static abstract class EmbeddedProjectMixin extends ProjectMixin {
```
Comment (rudra-superrr) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/JacksonCustomizations.java:253]: Remove tab and spaces.

File: backend/src/src-moderation/src/main/java/org/eclipse/sw360/moderation/ModerationHandler.java (hunk 1 above)
Comment (GMishx) [reply backend/src/src-moderation/src/main/java/org/eclipse/sw360/moderation/ModerationHandler.java]: Removed, please check.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/JacksonCustomizations.java (hunk 2 above)
Comment (GMishx) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/JacksonCustomizations.java]: Fixed, please check.

**Comment by rudra-superrr** [comment]
Code looks good.

**Comment by GMishx** [comment]
Note, the current implementation is missing SBOM changes in endpoint. And it is documented in the issue #1954

**Pull request merged**